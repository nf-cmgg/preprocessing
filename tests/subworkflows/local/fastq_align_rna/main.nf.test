nextflow_workflow {

    name "Test Workflow FASTQ_ALIGN_RNA"
    script "subworkflows/local/fastq_align_rna/main.nf"
    workflow "FASTQ_ALIGN_RNA"

    tag "subworkflows"
    tag "subworkflows/local"
    tag "subworkflows/local/fastq_align_rna"

    test("fastq align rna - star") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        sample_type:'RNA',
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("s3://test-data/genomics/homo_sapiens/illumina/fastq/test_R1.fastq.gz", checkIfExists: true),
                        file("s3://test-data/genomics/homo_sapiens/illumina/fastq/test_R2.fastq.gz", checkIfExists: true)
                    ],
                    "star", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/star/", checkIfExists: true),
                    []
                ])
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(
                workflow.out.bam.collect { it.collect { it instanceof Map ? it : file(it).name } },
                workflow.out.reports.collect { it.collect { it instanceof Map ? it : file(it).name } },
                workflow.out.versions
            ).match()
        }

    }

    test("fastq align rna - unknown aligner") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [
                        id:'test',
                        samplename:'test',
                        single_end:false,
                        sample_type:'RNA',
                        genome: [
                            fasta: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            fai: "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        ]
                    ], // meta map
                    [
                        file("s3://test-data/genomics/homo_sapiens/illumina/fastq/test_R1.fastq.gz", checkIfExists: true),
                        file("s3://test-data/genomics/homo_sapiens/illumina/fastq/test_R2.fastq.gz", checkIfExists: true)
                    ],
                    "woop", // aligner
                    file("s3://test-data/genomics/homo_sapiens/genome/star/", checkIfExists: true),
                    []
                ])
                """
            }
        }

        then {
            assert workflow.failed
            assert snapshot(workflow.stdout.findAll { it.contains("Unsupported") }).match()
        }

    }

}
