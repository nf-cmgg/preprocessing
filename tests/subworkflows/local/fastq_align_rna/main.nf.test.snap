{
    "fastq align rna - star": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "samplename": "test",
                        "single_end": false,
                        "sample_type": "RNA",
                        "genome": {
                            "fasta": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            "fai": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        }
                    },
                    "test.Aligned.out.bam"
                ]
            ],
            [
                [
                    {
                        "id": "test",
                        "samplename": "test",
                        "single_end": false,
                        "sample_type": "RNA",
                        "genome": {
                            "fasta": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            "fai": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        }
                    },
                    "test.Log.final.out"
                ],
                [
                    {
                        "id": "test",
                        "samplename": "test",
                        "single_end": false,
                        "sample_type": "RNA",
                        "genome": {
                            "fasta": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            "fai": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        }
                    },
                    "test.Log.out"
                ],
                [
                    {
                        "id": "test",
                        "samplename": "test",
                        "single_end": false,
                        "sample_type": "RNA",
                        "genome": {
                            "fasta": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna",
                            "fai": "s3://test-data/genomics/homo_sapiens/genome/seq/GCA_000001405.15_GRCh38_full_plus_hs38d1_analysis_set_chr21.fna.fai"
                        }
                    },
                    "test.Log.progress.out"
                ]
            ],
            [
                "versions.yml:md5,605896d89504a79cbbf1e0e450b15f00"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-28T16:12:33.555868971"
    },
    "fastq align rna - unknown aligner": {
        "content": [
            [
                "Unsupported aligner woop for sample test"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-05-28T16:17:08.089796673"
    }
}